src/encode.py

```diff
@@ -0,0 +1,52 @@
+import pandas as pd
+
+FEATURES_DEFAULT = ('lesson_id', 'exercise_code', 'learning_object', 'exercise_code_level')
+TIME_WINDOWS_DEFAULT = ('60s', '1h', '1d', '5d', '30d', '365d')
+COUNTERS_DEFAULT = ('attempts', 'wins')
+
+def encode_df(df, features=FEATURES_DEFAULT, counters = COUNTERS_DEFAULT, time_windows=TIME_WINDOWS_DEFAULT):
+    '''
+    Get the wanted counters of each student and dummifies the categorical variables of the dataset.
+    '''
+    df = add_counters_for_all_features(df, features, counters, time_windows)
+    df = pd.get_dummies(df, columns=['student_id', 'lesson_id', 'exercise_code', 'learning_object', 'exercise_code_level'])
```

Comment:
Why not reusing "features" here for the columns value? (with "student_id" to add)

Reply:
'student_id' is not a feature we want as a counter for now. I'm going to add test in the next PR